pkg/util/liveconfig/manager.go

```diff
@@ -35,11 +38,58 @@ type prod struct {
 	cachedCredentials    map[int]*rest.Config
 }
 
-func NewProd(location string, managedClustersClient containerservice.ManagedClustersClient) Manager {
-	return &prod{
+func NewProd(location string, managedClustersClient containerservice.ManagedClustersClient, log *logrus.Entry) Manager {
+	p := &prod{
 		location:              location,
 		managedClustersClient: managedClustersClient,
 		cachedCredentials:     make(map[int]*rest.Config),
 		hiveCredentialsMutex:  &sync.RWMutex{},
 	}
+	go p.hiveConfigRetrieveLoop(1, log)
```

Comment:
Having the index be passed in at this point may make it tricky when we move to using a dynamic index vs hardcoding to 1